alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,281 @@
+certificate expiration. Please refer to the next section for details.
+
+#### Expired SSL/TLS Certificates
+
+If the target uses SSL/TLS for communication, check if the SSL/TLS certificate
+has expired or certificate files accessible by Prometheus.
+
+The certificate used by Prometheus to scrape metrics endpoint is indicated in the
+`.spec.endpoints.tlsConfig.certFile` property of a `ServiceMonitor` or `PodMonitor`.
+The path of the certificate file points to a mounted volume on the Prometheus Pod.
+Therefore we can deduce which secret holds the certificate.
+
+Here is an example.
+We have a `ServiceMonitor` using `/etc/prometheus/secrets/metrics-client-certs/tls.crt`
+as its certificate file.
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/component: exporter
+    app.kubernetes.io/name: node-exporter
+    app.kubernetes.io/part-of: openshift-monitoring
+    app.kubernetes.io/version: 1.5.0
+    monitoring.openshift.io/collection-profile: full
+  name: node-exporter
+  namespace: openshift-monitoring
+spec:
+  endpoints:
+  - bearerTokenSecret:
+      key: ""
+    interval: 15s
+    port: https
+    relabelings:
+    - action: replace
+      regex: (.*)
+      replacement: $1
+      sourceLabels:
+      - __meta_kubernetes_pod_node_name
+      targetLabel: instance
+    scheme: https
+    tlsConfig:
+      ca: {}
+      caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+      cert: {}
+      certFile: /etc/prometheus/secrets/metrics-client-certs/tls.crt
+      keyFile: /etc/prometheus/secrets/metrics-client-certs/tls.key
+      serverName: node-exporter.openshift-monitoring.svc
+  jobLabel: app.kubernetes.io/name
+  namespaceSelector: {}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: exporter
+      app.kubernetes.io/name: node-exporter
+      app.kubernetes.io/part-of: openshift-monitoring
+```
+In the `Prometheus` that scrapes this target, the volume mounted at `/etc/prometheus/secrets/*`
+are the secrets having the same name as the subdirectory. In this example, the secret
+is `metrics-client-certs` in the namespace `openshift-monitoring`. Now we extract
+the certificate from the secret using this command:
+```bash
+oc extract secret/$SECRET_NAME -n $NAMESPACE --keys=tls.crt --to=- > certificate.crt
+```
+Then we inspect is expiration date.
+```bash
+openssl x509 -noout -enddate -in certificate.crt
+```
+The output should contain `notAfter` field as its expiration date.
+```bash
+notAfter=Aug  6 13:11:20 2025 GMT
+```
+
+Normally OpenShift **automatically renews** the certificates before expiration date.
+This date should be sometime in the future. If the certificate does expire without
+automatic renewal, please contact OpenShift support team.
+
+If the issue requires immediate resolution, please refer to [this guide on how to
+force a certificate renewal](https://access.redhat.com/solutions/5899121).
```

Comment:
it links to 3.11 KB, do we have 4.x KB?

Reply:
Though it is marked for  version 3.11. It works for 4.x, too. 

---------This is the content I used to put here --------------------
However, we can force to renew it manually if we need immediate resolution:
  1. Create a backup of the secret which contains the certificate.
```bash
oc get secret $SECRET_NAME -n $NAMESPACE -o yaml &> secret-tls.yaml
```
  2. Delete the secret holding the certificate.
```bash
oc delete secret $SECRET_NAME -n $NAMESPACE
```
  3. Check the newly created secret contains a valid certificate.
```bash
oc get secret $SECRET_NAME -n $NAMESPACE -o yaml
```
It should return certificate content with expiration date.
```bash
apiVersion: v1
data:
  tls.crt: ....
  tls.key: ....
kind: Secret
metadata:
  annotations:
    service.alpha.openshift.io/expiry: "2025-08-06T13:11:20Z"     <<< Here is the expiration date.
    service.beta.openshift.io/expiry: "2025-08-06T13:11:20Z"
    service.beta.openshift.io/originating-service-name: prometheus-k8s
    service.beta.openshift.io/originating-service-uid: a781e552-b3c3-4b74-b068-1d7152671b72
  name: prometheus-k8s-tls
  namespace: openshift-monitoring
  ownerReferences:
  - apiVersion: v1
    kind: Service
    name: prometheus-k8s
type: kubernetes.io/tls
```
  4. Restart the pods using this certificate by deleting the pods.
```bash
# by name
oc delete pod $POD_NAME -n $NAMESPACE
# or by label
oc delete pod -l $LABEL_SELECTOR -n $NAMESPACE
```